### PR TITLE
Improve astprinter

### PR DESCRIPTION
**Pull request by privat**
The module `astprinter` is used to print AST in order to debug synthetic statements and expressions.
It was quite limited.

This PR extends it to most Nit expressions and statements. The missing ones will be addressed in futures PR.

The PR also add a simple nit test program `syntax_expr.nit` that exhibits many syntactic constructions. It is used to test the astprinter through a test tool `test_astprinter` that (pretty?)-print given programs.

### CODE REVIEW

File: src/astprinter.nit [hunk 1]
```diff
@@ -90,10 +175,66 @@ redef class ABlockExpr
...
+	end
+end
+
+redef class AIfExpr
+	redef fun number_of_lines
 	do
-		v.write(value.to_s)
+		 #n_then.number_of_lines
+		 #n_else.number_of_lines
+		 return 3
```
Comment (dullin) [inline src/astprinter.nit:210]: Cleanup of comments?
Refactor magic number 3 to show explicit use of visit_block_noend else condition.

File: src/astprinter.nit [hunk 2]
```diff
@@ -90,10 +175,66 @@ redef class ABlockExpr
...
+		end
+	end
+end
+
+redef class AIfexprExpr
+	redef fun number_of_lines
+	do
+		 #n_then.number_of_lines
+		 #n_else.number_of_lines
+		 return 3
```
Comment (dullin) [inline src/astprinter.nit:229]: Same as AIfExpr.

File: src/astprinter.nit [hunk 3]
```diff
@@ -90,10 +175,66 @@ redef class ABlockExpr
...
+	end
+	redef fun accept_printer(v) do
+		v.write("if ")
+		v.enter_visit(n_expr)
+		v.write(" then ")
+		v.visit(n_then)
+		v.write(" else ")
+		v.visit(n_else)
```
Comment (dullin) [inline src/astprinter.nit:237]: Conditionnal on the else existing?

File: tests/sav/test_astprinter_args1.res [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+if b then a = 1
+d = a.as(not null)
+d = a.as(Int)
+do end
+do a = 1
+do
+	a = 1
+	a = 1
+end
+if b then else end
```
Comment (dullin) [inline tests/sav/test_astprinter_args1.res:46]: There's a pending else from probably AIfexprExpr always printing the else.

File: tests/syntax_expr.nit [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+if b then else end
+if b then else a = 8
+if b then else
+	a = 9
+	a = 9
+end
+
+a = if b then 1 else 2
+a = if b then if b then 3 else 4 else if b then 5 else 6
+
```
Comment (dullin) [inline tests/syntax_expr.nit:91]: Add a nested control structure like

```nit
if a == 1 then
	if a == 2 then
		a = 10
	end
end
```

**Review by dullin — CHANGES_REQUESTED** [review]
* In a if expression block, the else is always printed even if not present.
* Magic number 3 for number of lines to make more explicit.

File: src/astprinter.nit (hunk 3 above)
Comment (privat) [reply src/astprinter.nit:247]: this is for the ternary operator, there is always an else part

File: tests/sav/test_astprinter_args1.res (hunk 4 above)
Comment (privat) [reply tests/sav/test_astprinter_args1.res]: for some reason, the else is in the ast but is empty. So I just do not print empty else parts

File: tests/syntax_expr.nit (hunk 5 above)
Comment (privat) [reply tests/syntax_expr.nit:102]: added

**Pull request merged**